Client/Assets/WindowCloseHelper.js

```diff
@@ -3,9 +3,70 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 (function () {
-  var _close = window.close;
-  window.close = function () {
-    webkit.messageHandlers.windowCloseHelper.postMessage(null);
-    _close();
-  };
+    if (!window.__firefox__) {
+        window.__firefox__ = {};
+    }
+    if (!window.__firefox__.messages) {
+        window.__firefox__.messages = {};
+    }
+    window.__firefox__.messages.close = "FIREFOX_MESSAGE_CLOSE";
+ 
+    var _close = window.close;
+ 
+    window.close = function () {
+        webkit.messageHandlers.windowCloseHelper.postMessage(null);
+        _close();
```

Comment:
While here, might as well wrap this properly and use `_close.apply(this, arguments);` like you did below.